### PR TITLE
result: null

### PR DESCRIPTION
**Pull request by niulingyun**

### CODE REVIEW

File: jsonrpc.go [hunk 1]
```diff
@@ -451,11 +451,16 @@ func (rpcResponse *RPCResponse) GetObject(toType interface{}) error {
 		return err
 	}
 
-	err = json.Unmarshal(js, toType)
+	err = json.Unmarshal(js, &toType)
```
Comment (ybbus) [inline jsonrpc.go]: why did you add the operator here? isnt interface{} always a pointer?

Comment (niulingyun) [reply jsonrpc.go]: yes, when the result is "null", `toType` can be `nil`, rather than a `struct` or something.

Comment (ybbus) [reply jsonrpc.go]: please add a test that is only green with this fix.

Comment (ybbus) [reply jsonrpc.go]: @niulingyun 
Sorry for the delay, I see what's your point, and you are correct.

I will take a closer look e.g. under which circumstances this fix would break client code.

But it seems reasonable to merge this PR

File: jsonrpc.go [hunk 2]
```diff
@@ -451,11 +451,16 @@ func (rpcResponse *RPCResponse) GetObject(toType interface{}) error {
...
+
 	if err != nil {
 		return err
 	}
 
+	if toType == nil {
```
Comment (ybbus) [inline jsonrpc.go]: This can only happen if you provide a nil pointer as input. But then you should also expect a nil as ouput that is the go aquivalent of json null.

I don't know if this should be an error since:
1. the request you send to the rpc server was successfull
2. the response was null
This seems ok and should not throw an error, should it?

Comment (ybbus) [reply jsonrpc.go]: Consider this:

```go
p := &Person{}
err := result.GetObject(p)
```

You won't get an error, even if the response from the server was:
`{"jsonrpc":"2.0","result":null,"id":1}`

**Comment by ybbus** [comment]
I think the function GetObject() should exactly behave like json.Unmarshal().
And this is not the case when I merge this PR.
